File: TicTacToe C++/TicTacToe.cpp
#include <iostream>
#include <vector>

using namespace std;

// Function to print the Tic Tac Toe board
void printBoard(const vector<char>& board) {
    cout << "-------------" << endl;
    for (int i = 0; i < 9; ++i) {
        cout << "| " << board[i] << " ";
        if ((i + 1) % 3 == 0) {
            cout << "|" << endl;
            cout << "-------------" << endl;
        }
    }
}

// Function to check if a player has won
bool checkWin(const vector<char>& board, char player) {
    // Check rows
    for (int i = 0; i < 9; i += 3) {
        if (board[i] == player && board[i + 1] == player && board[i + 2] == player) {
            return true;
        }
        
    }

    // Check columns
    for (int i = 0; i < 3; ++i) {
        if (board[i] == player && board[i + 3] == player && board[i + 6] == player) {
            return true;
        }
    }

    // Check diagonals
    if ((board[0] == player && board[4] == player && board[8] == player) ||
        (board[2] == player && board[4] == player && board[6] == player)) {
        return true;
    }

    return false;
}

// Function to check if the board is full (draw)
bool checkDraw(const vector<char>& board) {
    for (char cell : board) {
        if (cell == ' ') {
            return false; // Found an empty cell, not a draw
        }
    }
    return true; // No empty cells, it's a draw
}

int main() {
    vector<char> board(9, ' '); // Initialize a 9-element vector with spaces
    char currentPlayer = 'X';
    int move;

    cout << "Welcome to Tic Tac Toe!" << endl;

    do {
        printBoard(board);
        cout << "Player " << currentPlayer << ", enter your move (1-9): ";
        cin >> move;

        // Validate the move
        if (move >= 1 && move <= 9 && board[move - 1] == ' ') {
            board[move - 1] = currentPlayer;

            // Check for win
            if (checkWin(board, currentPlayer)) {
                printBoard(board);
                cout << "Player " << currentPlayer << " wins!" << endl;
                break;
            }

            // Check for draw
            if (checkDraw(board)) {
                printBoard(board);
                cout << "It's a draw!" << endl;
                break;
            }

            // Switch players
            currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';
        } else {
            cout << "Invalid move. Please try again." << endl;
        }

    } while (true);

    return 0;
}

File: TicTacToe Python/TicTacToe.py
def print_board(board):
    for row in board:
        print(" | ".join(row))
        print("-" * 9)

def check_win(board, player):
    for i in range(3):
        if all([board[i][j] == player for j in range(3)]) or \
           all([board[j][i] == player for j in range(3)]):
            return True
    if all([board[i][i] == player for i in range(3)]) or \
       all([board[i][2 - i] == player for i in range(3)]):
        return True
    return False

def tic_tac_toe():
    board = [[" " for _ in range(3)] for _ in range(3)]
    player1 = "X"
    player2 = "O"
    current_player = player1
    moves = 0

    print("Tic-Tac-Toe Game")
    print_board(board)

    while moves < 9:
        try:
            row = int(input(f"Player {current_player}, enter row (0-2): "))
            col = int(input(f"Player {current_player}, enter column (0-2): "))
        except ValueError:
            print("Invalid input. Please enter numbers between 0 and 2.")
            continue
        
        if 0 <= row <= 2 and 0 <= col <= 2 and board[row][col] == " ":
            board[row][col] = current_player
            print_board(board)
            moves += 1

            if check_win(board, current_player):
                print(f"Player {current_player} wins!")
                break

            current_player = player2 if current_player == player1 else player1
        else:
            print("Invalid move. Try again.")

    if moves == 9 and not check_win(board, player1) and not check_win(board, player2):
        print("It's a tie!")

tic_tac_toe()